### PR TITLE
[REVIEW] Fix boolean indexing resulting in a single row

### LINKED ISSUE #2339: [BUG] loc fails with filtering by column value
I am trying to select rows on the basis of a column value. I see the error: `ValueError: Length mismatch: Expected axis has 1 elements, new values have 2 elements` when I try to run the following code: 

```
import cudf
df = cudf.DataFrame({"a":["xyz","abc"],"b":["t1","t2"]})
df = df.loc[df['b']=="t2"]
```

I can use ```df = df[df['b']=="t2"]``` for now, though, but would love to have the loc functionality, if possible.

### PR DESCRIPTION
**Pull request by shwina**
Band-aid solution that fixes #2339

### CODE REVIEW

File: python/cudf/cudf/core/indexing.py [hunk 1]
```diff
@@ -120,19 +120,24 @@ def _can_downcast_to_series(self, df, arg):
...
+            else:
+                # row selection using boolean indexing - never downcasts
+                if pd.api.types.is_bool_dtype(as_column(arg[0]).dtype):
+                    return False
             dtypes = df.dtypes.values.tolist()
             all_numeric = all(
                 [pd.api.types.is_numeric_dtype(t) for t in dtypes]
             )
-            all_identical = dtypes.count(dtypes[0]) == len(dtypes)
-            if all_numeric or all_identical:
```
Comment (shwina) [inline python/cudf/cudf/core/indexing.py]: We don't handle the `all_identical` case. We can't transpose a row of strings.

File: python/cudf/cudf/core/indexing.py [hunk 2]
```diff
@@ -217,7 +223,11 @@ def _getitem_tuple_arg(self, arg):
                     start = self._df.index[0]
                 df.index = as_index(start)
             else:
-                df.index = as_index(arg[0])
+                row_selection = column.as_column(arg[0])
+                if pd.api.types.is_bool_dtype(row_selection.dtype):
+                    df.index = self._df.index.take(row_selection)
+                else:
+                    df.index = as_index(row_selection)
```
[the remark this answers is not in the document]
Comment (shwina) [reply python/cudf/cudf/core/indexing.py]: Here `row_selection` is the actual values (not positions) of the index, so we just coerce it into an `Index`.

... except in the case of boolean masking, where `row_selection` is just a boolmask. So we apply that boolmask onto the index to get the resulting index.

[the remark this answers is not in the document]
Comment (kkraus14) [reply python/cudf/cudf/core/indexing.py]: Got it, thanks for explaining

**Comment by thomcom** [comment]
lgtm

**Pull request merged**